src/samples/coap/simple-server.c

```diff
@@ -237,7 +237,7 @@ main(int argc, char *argv[])
         return -1;
     }
 
-    if (!sol_coap_server_register_resource(server, &light, NULL)) {
+    if (sol_coap_server_register_resource(server, &light, NULL) < 0) {
         SOL_WRN("Could not register resource for the light");
```

Comment:
I know this is not in the patch but wouldn't this be better ? 

s/Could not register resource for the light/Could not register light resource/g